### PR TITLE
gh-pages for the repo

### LINKED ISSUE #177: Create Github pages for this repo
Docusaurus could be used.
Example: https://shard-labs.github.io/starknet-devnet/

### PR DESCRIPTION
**Pull request by shramee**
## Usage related changes

- Added `./www` with docusaurus stuff, deploy to `gh-pages`
- Closes #177 
- Preview at https://shramee.github.io/starknet-hardhat-plugin/

## Development related changes

- Added `./www` with docusaurus stuff.
- Changes in [README.md](https://github.com/Shard-Labs/starknet-hardhat-plugin/blob/master/README.md) should be updated in `./www/docs/intro.md`
- Changes in [README-dev.md](https://github.com/Shard-Labs/starknet-hardhat-plugin/blob/master/README-dev.md) should be updated in `./www/docs/dev.md`
- Deployment:
```bash
cd www
npm install
USE_SSH=true npm run deploy
```

## Checklist:

-   [x] Formatted the code
-   [x] No linter errors + tried to avoid introducing linter warnings
-   [x] Performed a self-review of the code
-   [x] Rebased to the last commit of the target branch (or merged it into my branch)
-   [x] Documented the changes
    -   [ ] If `src/type-extensions.ts` was changed, updated the line number referencing this file in the `## API` section of README.md
-   [ ] Updated the `test` directory (with a test case consisting of `network.json`, `hardhat.config.ts`, `check.ts`)
-   [x] Linked issues which this PR resolves
-   [ ] Created a PR to the `plugin` branch of [`starknet-hardhat-example`](https://github.com/Shard-Labs/starknet-hardhat-example):
    -   < EXAMPLE_REPO_PR_URL > 
    -   [ ] Modified `test.sh` to use my example repo branch
    -   [ ] Restored `test.sh` to to use the original branch (after the example repo PR has been merged)
-   [ ] All tests are passing (for external contributors who don't have access to the CI/CD pipeline)

### CODE REVIEW

**Comment by shramee** [comment]
#177 gh-pages with Docusaurus.

**Review by FabijanC — COMMENTED** [review]
Thanks, looks great! Made a few comments. Once merged, I will edit our current README.md to include a link to the page.

File: www/src/pages/markdown-page.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
```
[the remark this answers is not in the document]
Comment (FabijanC) [reply www/src/pages/markdown-page.md:1]: Ok, rechecked, it's ok to keep this.

File: package.json [hunk 2]
```diff
@@ -29,7 +29,8 @@
         "build": "rm -rf dist && tsc && npm run copy-files",
         "copy-files": "cp src/*.py dist/src/",
         "lint": "eslint $(git ls-files '*.ts')",
-        "format": "prettier-eslint --list-different --write $(git ls-files '*.ts')"
+        "format": "prettier-eslint --list-different --write $(git ls-files '*.ts')",
+        "formatjs": "prettier-eslint --list-different --write $(git ls-files '*.js')"
```
[the remark this answers is not in the document]
Comment (shramee) [reply package.json]: Added 'em all `'*.ts' '*.js' '*.json'` :wink:

[the remark this answers is not in the document]
Comment (shramee) [reply package.json]: dd38b5aea8c22cfa76414c51cc6aa20f670cafa4

[the remark this answers is not in the document]
Comment (FabijanC) [reply package.json]: I tried running `npm run format` and it doesn't work now. Also, some json files we want to stay minified. The best is to keep .js and .ts as targets of linting and formatting. But to speed up the development process, I will merge this and do the quick-fix myself.

**Pull request merged**